backend/cmd/headlamp.go

```diff
@@ -922,11 +1132,14 @@ func (c *HeadlampConfig) deleteCluster(w http.ResponseWriter, r *http.Request) {
 }
 
 func (c *HeadlampConfig) addClusterSetupRoute(r *mux.Router) {
-	// We do not support this feature when in-cluster
+	// Do not add the route if dynamic clusters are disabled
 	if !c.enableDynamicClusters {
 		return
 	}
+	// Get stateless cluster
+	r.HandleFunc("/parseKubeConfig", c.parseKubeConfig).Methods("POST")
```

Comment:
I think it make sense. Since it is taking kubeconfig as in input. And then it parses the base64 encoded kubeconfig and returns it in desired format. It is not replacing any existing config. It is just parsing the values and returning it to frontend.